supertokens_python/recipe/session/interfaces.py

```diff
@@ -327,3 +451,100 @@ def sync_update_session_data(
     # This is there so that we can do session["..."] to access some of the members of this class
     def __getitem__(self, item: str):
         return getattr(self, item)
+
+
+# Session claims:
+_T = TypeVar("_T")
+JSONObject = Dict[str, Any]
+
+JSONPrimitive = Union[str, int, bool, None, Dict[str, Any]]
+
+FetchValueReturnType = Union[_T, None]
+
+
+class SessionClaim(ABC, Generic[_T]):
+    def __init__(self, key: str) -> None:
+        self.key = key
+
+    def fetch_value(
+        self, user_id: str, user_context: Optional[Dict[str, Any]] = None
+    ) -> MaybeAwaitable[Optional[_T]]:
+        pass
+
+    @abstractmethod
+    def add_to_payload_(
+        self,
+        payload: JSONObject,
+        value: _T,
+        user_context: Union[Dict[str, Any], None] = None,
+    ) -> JSONObject:
+        """Saves the provided value into the payload, by cloning and updating the entire object"""
+
+    @abstractmethod
+    def remove_from_payload_by_merge_(
+        self, payload: JSONObject, user_context: Dict[str, Any]
+    ) -> JSONObject:
+        """Removes the claim from the payload by setting it to null, so mergeIntoAccessTokenPayload clears it"""
+
+    @abstractmethod
+    def remove_from_payload(
+        self, payload: JSONObject, user_context: Dict[str, Any]
+    ) -> JSONObject:
+        """Gets the value of the claim stored in the payload
+
+        Returns:
+            JSONObject: Claim value
+        """
+
+    @abstractmethod
+    def get_value_from_payload(
+        self, payload: JSONObject, user_context: Union[Dict[str, Any], None] = None
+    ) -> Union[_T, None]:
+        pass
+
+    async def build(
+        self, user_id: str, user_context: Union[Dict[str, Any], None] = None
+    ) -> JSONObject:
+        value = await resolve(self.fetch_value(user_id, user_context))
+
+        if value is None:
+            return {}
+
+        return self.add_to_payload_({}, value, user_context)
+
+
+# class ClaimValidationResultValid:
+#     def __init__(self):
+#         self.is_valid = True
+#
+# class ClaimValidationResultInvalid:
+#     def __init__(self, reason: str):
+#         self.is_valid = False
+#         self.reason = reason
+#
+# ClaimValidationResult = Union[ClaimValidationResultValid, ClaimValidationResultInvalid]
+
+
+ClaimValidationResult = Dict[str, Any]
```

Comment:
I guess this is a TODO?

Reply:
`ClaimValidationResult = Dict[str, Any]` this one? No. I'm keeping it as a dict. Just giving it an alias (`ClaimValidationResult`) for code readability..

Reply:
I meant that a few lines above it has a different definition, or it just looked a bit confusing for me.

Reply:
Ok. Will remove those comments.